Directory.Build.props

```
@@ -54,6 +54,7 @@
     <NetCoreAppCurrentBrandName>.NET $(NetCoreAppCurrentVersion)</NetCoreAppCurrentBrandName>
     <NetCoreAppCurrent>net$(NetCoreAppCurrentVersion)</NetCoreAppCurrent>
     <NetCoreAppMinimum>net6.0</NetCoreAppMinimum>
+    <NetCoreAppMinimum Condition="'$(DotNetBuildFromSource)' == 'true'">net7.0</NetCoreAppMinimum>
```

Use the msbuild property so that this works for future releases as well.

```suggestion
    <NetCoreAppMinimum Condition="'$(DotNetBuildFromSource)' == 'true'">$(NetCoreAppCurrent)</NetCoreAppMinimum>
```

---

@rainersigwald do you know if there are any downsides in having duplicate target framework entries in the `TargetFrameworks` property? I assume that the entries are de-duplicated?

---

They are sorta-kinda accidentally deduplicated. Specifically, the MSBuild engine collapses N requests for `foo.csproj with TargetFramework=net7.0` into one and returns cached results, but might confuse other systems. It doesn't appear to confuse VS, though, in my quickie test. So I don't think it's a big deal but might cause some minor problems down the road.